digital_benchmark/twitter_benchmark/views.py

```diff
@@ -32,9 +37,74 @@ class SuccessView(View):
     def get(self, request):
         obj = DataProvider(request.GET['oauth_verifier'], request.GET['oauth_token'])
         # for testing
+        #ehmadzubair
         data = obj.get_other_tweet('ehmadzubair')
-        #data = TwitterDataParser.parser_other_tweet(data=data)
-        print(data)
+        data = TwitterDataParser.parser_other_tweet(data)
+        #print(data)
         return render(request, 'success/index.html')
 
 
+class UserDataList(generics.RetrieveAPIView):
+    permission_classes = [IsAuthenticated]
+
+    def retrieve(self, request, pk=None):
+        queryset = UserData.objects.filter(app_user=self.request.user)
+        serializer = UserDataSerializer(queryset, many=True)
+        return Response(serializer.data)
+
+
+class UserTweetList(generics.ListAPIView):
+    permission_classes = [IsAuthenticated]
+    serializer_class = UserTweetSerializer
+
+    def get_queryset(self):
+        return UserTweet.objects.filter(app_user_id=self.request.user.id)
+
+
+class OtherTweetList(generics.ListAPIView):
+    permission_classes = [IsAuthenticated]
+    serializer_class = OtherTweetSerializer
+
+    def get_queryset(self):
+        return OtherTweet.objects.filter(screen_name=self.kwargs.get('screenname'))
+
+
+class UserTweetByIdList(generics.ListAPIView):
+    permission_classes = [IsAuthenticated]
+    serializer_class = UserTweetSerializer
+
+    def get_queryset(self):
+        return UserTweet.objects.filter(tweet_id=self.kwargs.get('id')).order_by('-created_at')
+
+
+class UserLatestTweet(generics.RetrieveAPIView):
+    permission_classes = [IsAuthenticated]
+
+    def retrieve(self, request, id):
+        try:
```

Comment:
override "get_queryset()" instead of "retrieve()". Read the difference between two methods.